### PR TITLE
Enhancement: Add a new GUI Event:  UI_WINDOW_RESIZED

### PR DESCRIPTION
**Pull request by JordanKarp**
When creating UIWindows, I find I want to know if the user has resized the window. It currently looks like there are only two events related to UIWindow: UI_WINDOW_CLOSE, UI_WINDOW_MOVED_TO_FRONT, neither of which check for a resize. However UIWindows themselves handle resizing perfectly.

So this enhancement is to create a new GUI event: UI_WINDOW_RESIZED that is fired when a resize occurs.

I've added the new constant and updated the process_events() function to create a new GUI event and post it upon resizing. 

Documentation: 
> Fired when a window is resized.
> 
> 'type' : pygame_gui.UI_WINDOW_RESIZED,
> 'ui_element' : The UIWindow that fired this event.
> 'ui_object_id' : The most unique ID for the element that fired this event.
> 

Some example usage:
```
for event in pygame.event.get():
     if event.type == pygame_gui.UI_WINDOW_RESIZED:
         if event.ui_element == window:
             print("Window resized")
```

Let me know what else is needed!

### CODE REVIEW

**Comment by MyreMylar** [comment]
I think you  missing the `pygame_gui/__init__.py` bits that project the constant into the `pygame_gui.` namespace for users - so you can do `pygame_gui.UI_WINDOW_RESIZED`. I think at the minute you would have to do  `pygame_gui._constants.UI_WINDOW_RESIZED`

At the moment you have the event firing off once when the user starts resizing the window. Is that what you want it to do? 

You could have it firing off either when the resizing process ends, or every time the window changes dimensions (see: `.set_dimensions()`) - I think this is what a pygame/OS level window does but it might be worth checking. The pygame events are called something like: `WINDOWEVENT_RESIZED` and `WINDOWEVENT_SIZE_CHANGED` and come from the underlying SDL. There is also an older event called `VIDEORESIZE` IIRC. Anyway might be worth checking their behaviour with a resizable pygame window as we probably want to stay relatively close to that behaviour in pygame_gui.

Finally there is tests and documentation. Event docs are in `docs/source/events.rst` and tests for the window are in `tests/test_elements/test_ui_window.py`

**Comment by JordanKarp** [comment]
Ah thank you! I appreciate your guidance.

I believe I've fixed the ```__init__``` bits and I've updated the event documentation. Also fixed a small error in ```UI_WINDOW_MOVED_TO_FRONT``` event docs.

You are correct in that I wanted this to fire once the resizing was completed. I moved ```UI_WINDOW_RESIZED```  event post to the ```.set_dimensions()``` function and have updated that test. I had a hard time figuring out how ```pygame-ce``` does it and I'm not entirely sure this is the best spot for the event to fire. I'm happy to adjust it if you have a better idea!

File: pygame_gui/elements/ui_window.py [hunk 1]
```diff
@@ -131,6 +131,10 @@ def set_dimensions(self, dimensions: Union[pygame.math.Vector2,
                 container_pos = (self.relative_rect.x + self.shadow_width,
                                  self.relative_rect.y + self.shadow_width)
                 self._window_root_container.set_relative_position(container_pos)
+                window_resize_event = pygame.event.Event(UI_WINDOW_RESIZED,
+                                        {'ui_element': self,
+                                         'ui_object_id': self.most_specific_combined_id})
```
Comment (MyreMylar) [inline pygame_gui/elements/ui_window.py:136]:
```suggestion
                                         'ui_object_id': self.most_specific_combined_id,
                                         'size': new_container_dimensions})
```

File: tests/test_elements/test_ui_window.py [hunk 2]
```diff
@@ -81,11 +81,22 @@ def test_set_dimensions(self, _init_pygame, default_ui_manager: IUIManagerInterf
...
                                         window.get_container().rect.top + 10)
         assert button.rect.topright == (474, 253)
+        
+        confirm_event_fired = False
+        event_object_id = None
+        for event in pygame.event.get():
+            if (event.type == pygame_gui.UI_WINDOW_RESIZED and
+                    event.ui_element == window):
+                confirm_event_fired = True
+                event_object_id = event.ui_object_id
```
Comment (MyreMylar) [inline tests/test_elements/test_ui_window.py:97]:
```suggestion
                event_object_id = event.ui_object_id
                new_dimensions = event.size
```

File: tests/test_elements/test_ui_window.py [hunk 3]
```diff
@@ -81,11 +81,22 @@ def test_set_dimensions(self, _init_pygame, default_ui_manager: IUIManagerInterf
...
+        assert confirm_event_fired
+        assert event_object_id == 'window'
```
Comment (MyreMylar) [inline tests/test_elements/test_ui_window.py:99]:
```suggestion
        assert event_object_id == 'window'
        assert new_dimensions == (300, 400)
```

File: docs/source/events.rst [hunk 4]
```diff
@@ -319,9 +319,28 @@ clicked on by a user.
...
+                print("Window moved to front")
+
+:class:`UIWindow <pygame_gui.elements.UIWindow>` - UI_WINDOW_RESIZED
+..................................................................
+
+Fired when a window is resized.
+
+ - **'type'** : pygame_gui.UI_WINDOW_RESIZED,
+ - **'ui_element'** : The :class:`UIWindow <pygame_gui.elements.UIWindow>` that fired this event.
+ - **'ui_object_id'** : The most unique ID for the element that fired this event.
```
Comment (MyreMylar) [inline docs/source/events.rst:333]:
```suggestion
 - **'ui_object_id'** : The most unique ID for the element that fired this event.
 - **'size'** : The new size of the window.
```

**Comment by MyreMylar** [comment]
OK great, I added a couple of attributes to the event to show the new 'external' and 'internal' size of the window after this resizing.

Hopefully this all works for your use case. Thanks for the PR!

**Review by MyreMylar — APPROVED** [review]
👍

**Pull request merged**